module/state_synchronization/indexer/indexer.go

```diff
@@ -161,14 +161,16 @@ func (i *Indexer) onBlockIndexed() error {
 		// we need loop here because it's possible for a height to be missed here,
 		// we should guarantee all heights are processed
 		for height := lastProcessedHeight + 1; height <= highestIndexedHeight; height++ {
-			header, err := i.indexer.headers.ByHeight(height)
```

Comment:
This call might cause 30% CPU spikes on startup for hours, if there are lots of height to iterate.

We are not using the header, I don't think it's necessary to query it.